src/protect-function-app/greeting.cs

```diff
@@ -0,0 +1,49 @@
+/*
+This is an Azure Function that responds at GET /api/greeting.
+
+Using the built-in authentication and authorization capabilities (sometimes
+referred to as "Easy Auth") of Azure Functions, offloads part of the authentication
+and authorization process by ensuring that every request to this Azure Function has
+an access token. That access token has had its signature, issuer (iss), expiry
+dates (exp, nbf), and audience (aud) validated. This means all that is left to
+perform is any per-function authorization related to your application.
+*/
+
+using System;
+using System.IO;
+using System.Security.Claims;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+
+namespace Api
+{
+    public static class Greeting
+    {
+        /*
+        Azure Functions HTTP Triggers perform automatic input binding of the
+        Easy Auth-validated JWT token data into the ClaimsPincipal.
+        Using the claims principal allows additional access token validation specific to this function.
+        */
+        [FunctionName("greeting")]
+        public static IActionResult Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest req,
+            ClaimsPrincipal principal)
+        {
+            // This API endpoint requires the "Greeting.Read" scope to be present, if it is
+            // not, then reject the request with a 403.
+            if (!principal.Claims.Any(
+                  c => c.Type == "http://schemas.microsoft.com/identity/claims/scope"
+                  && c.Value.Split(' ').Contains("Greeting.Read")))

```

Comment:
Instead of both an `Any` on the claims and a `Contains` on the scope string, should we use instead a `Contains` on the claims collection and `String.Equals` on the scope name to ensure we match _only_ `Greeting.Read`?

I haven't run it, but my Spidey Sense tells me this would match both `Greeting.Read` and `Greeting.Read.Deny`.

Reply:
Contains on a string array is a case-sensitive , exact match search. "Greeting.Read.Blah" will not be found for a Contains of "Greeting.Read" - no partial match for that. That's different than string.Contains. we moved specifically to this implementation for this exact reason you mentioned.

Reply:
Hey @mmacy, if it makes you feel and better, I tested it locally before I made the recommendation to Fernando; just to verify it didn't have the problem you were mentioning. Your spidy sense was the same I had. :) Rusty, nah ... cautious, yes!